### PR TITLE
Use cobra to parse CLI

### PR DESCRIPTION
**Pull request by myidpt**
Use Cobra to parse CLI.
This fixes issue #226.
```release-note
Use cobra to parse CLI
```

### CODE REVIEW

**Comment by lookuptable** [comment]
Thanks for taking care of this. 

This PR closes #226

File: cmd/node_agent/main.go [hunk 1]
```diff
@@ -15,33 +15,45 @@
...
-		"Node identity private key file")
-	naConfig.IstioCAAddress = flag.String("ca-address", "127.0.0.1",
-		"Istio CA address")
-	naConfig.ServiceIdentityDir = flag.String("cert-dir", "./", "Certificate directory")
-	naConfig.RootCACertFile = flag.String("root-cert", "", "Root Certi file")
-	naConfig.Env = flag.Int("env", na.ONPREM, "Node Environment : onprem | gcp")
+	flags := rootCmd.Flags()
+
+	flags.StringVar(&naConfig.ServiceIdentity, "service-identity", "", "Service Identity the node agent is managing")
+	flags.StringVar(&naConfig.ServiceIdentityOrg, "org", "Juju org", "organization for the cert")
```
Comment (lookuptable) [inline cmd/node_agent/main.go]: nit: s/organization/Organization/

File: cmd/node_agent/main.go [hunk 2]
```diff
@@ -15,33 +15,45 @@
...
+	flags.IntVar(&naConfig.RSAKeySize, "key-size", 1024, "Size of generated private key")
+	flags.StringVar(&naConfig.NodeIdentityCertFile, "na-cert", "", "Node Agent identity cert file")
+	flags.StringVar(&naConfig.NodeIdentityPrivateKeyFile, "na-key", "", "Node identity private key file")
+	flags.StringVar(&naConfig.IstioCAAddress, "ca-address", "127.0.0.1", "Istio CA address")
+	flags.StringVar(&naConfig.ServiceIdentityDir, "cert-dir", "./", "Certificate directory")
+	flags.StringVar(&naConfig.RootCACertFile, "root-cert", "", "Root Certi file")
```
Comment (lookuptable) [inline cmd/node_agent/main.go]: s/Certi/certificate/

File: cmd/node_agent/na/nodeagent.go [hunk 3]
```diff
@@ -171,9 +171,9 @@ func (na *nodeAgentInternal) sendCSR() ([]byte, *pb.Response, error) {
 		glog.Errorf("Cannot construct the dial options with error %v", err)
 		return nil, nil, err
 	}
-	conn, err := grpc.Dial(*na.config.IstioCAAddress, dialOptions...)
+	conn, err := grpc.Dial(na.config.IstioCAAddress, dialOptions...)
 	if err != nil {
-		glog.Fatalf("Failed ot dial %s: %v", *na.config.IstioCAAddress, err)
+		glog.Fatalf("Failed ot dial %s: %v", na.config.IstioCAAddress, err)
```
Comment (lookuptable) [inline cmd/node_agent/na/nodeagent.go]: nit: s/ot/to/

File: cmd/node_agent/na/nodeagent.go [hunk 4]
```diff
@@ -43,31 +43,31 @@ const (
...
 
 	// Node Identity certificate file
-	NodeIdentityCertFile *string
+	NodeIdentityCertFile string
 
 	// Service Identity
-	ServiceIdentity *string
+	ServiceIdentity string
 
 	// Service Identity
```
Comment (lookuptable) [inline cmd/node_agent/na/nodeagent.go]: Organization for service identity.

File: cmd/node_agent/na/nodeagent.go (hunk 3 above)
Comment (myidpt) [inline cmd/node_agent/na/nodeagent.go]: Done.

File: cmd/node_agent/na/nodeagent.go (hunk 4 above)
Comment (myidpt) [inline cmd/node_agent/na/nodeagent.go]: Done.

File: cmd/node_agent/main.go (hunk 2 above)
Comment (myidpt) [inline cmd/node_agent/main.go]: Done.

File: cmd/node_agent/main.go (hunk 1 above)
Comment (myidpt) [inline cmd/node_agent/main.go]: Done.

**Comment by myidpt** [comment]
That's OK. I have fixed these nit issues in this PR. They are easy to fix :)

**Pull request merged**